src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,144 @@
+import { useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person, SearchParameters, SortValues } from '../types';
+import { getPeople } from '../api';
+import { getPreparedPersons } from '../utils/getPreparedPersons';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isErrorToGetPeople, setIsErrorToGetPeople] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get(SearchParameters.Sort) as SortValues || '';
+  const order = searchParams.get(SearchParameters.Order) || '';
+  const centuries = searchParams.getAll(SearchParameters.Centuries) || [];
+  const query = searchParams.get(SearchParameters.Query) || '';
+  const sex = searchParams.get(SearchParameters.Sex) || '';
+
+  const isNotPeopleOnServer
+    = !people.length
+    && !isLoading
+    && !isErrorToGetPeople;
+
+  const isErrorMessageVisible = isErrorToGetPeople && !isLoading;
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then((allPersons) => {
+        const PeopleWithRelatives = getPreparedPersons(allPersons);
+
+        setPeople(PeopleWithRelatives);
+      })
+      .catch(() => {
+        setIsErrorToGetPeople(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const sortPeople = (sortType: SortValues,
+    isOrder: string,
+    allPeople: Person[]) => {
+    const copyAllPeople = [...allPeople];
+
+    if (sortType) {
+      copyAllPeople.sort((a, b) => {
+        switch (sortType) {
+          case SortValues.Name:
+            return a.name.localeCompare(b.name);
+          case SortValues.Sex:
+            return a.sex.localeCompare(b.sex);
+          case SortValues.Born:
+            return a.born - b.born;
+          case SortValues.Died:
+            return a.died - b.died;
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (isOrder === 'desc') {
+      copyAllPeople.reverse();
+    }
+
+    return copyAllPeople;
+  };
+
+  function filterPeople(allPeople: Person[],
+    queryFilter: string,
+    filterBySex: string,
+    filterByCenturies: string[]) {
+    let copyAllPeople = [...allPeople];
```

Comment:
The `filter` method returns a copy of the array, so you don't need to do it manually.

Reply:
if (queryFilter) {
      allPeople = allPeople.filter(man => man.name.toLowerCase()
        .includes(queryFilter.toLowerCase()));
    }
    
   then I will modify the input array